main.py

```diff
@@ -297,17 +297,19 @@ def eval_worker(gpu, ngpu_per_node, args, res_queue=None):
 
 
 def writer(args, res_queue):
-    result_path = os.path.join(args.exp_dir, args.result_fname)
-    myprint("Writer process standing by... %s" % result_path, color='yellow')
 
     # We need dimensions of the inference dataset.
     if args.infer:
         files_path = args.infer_files
     else:
         assert False, "writer called but infer = False. Not sure what file to write?"
+
     total_size = 0
     write_dimension = 0
     for fname in files_path:
+        prefix = os.path.basename(fname).split(".")[0]
+        result_path = os.path.join(args.exp_dir, prefix + "_" + args.result_fname)
```

Comment:
I think the `prefix` is used to construct the `exp_dir` directory already?

Reply:
I don't think it is. I double checked. It overwrites the same files multiple times if I don't add this prefix.

Reply:
Yes, I see now. I was confusing it with another `prefix` from a different portion of the code. This looks good.